be/src/olap/rowset/segment_v2/binary_dict_page.cpp

```diff
@@ -207,71 +200,54 @@ Status BinaryDictPageDecoder::init() {
         // copy the codewords into a temporary buffer first
         // And then copy the strings corresponding to the codewords to the destination buffer
         TypeInfo* type_info = get_scalar_type_info(OLAP_FIELD_TYPE_INT);
-
         RETURN_IF_ERROR(ColumnVectorBatch::create(0, false, type_info, nullptr, &_batch));
-        _data_page_decoder.reset(_bit_shuffle_ptr = new BitShufflePageDecoder<OLAP_FIELD_TYPE_INT>(_data, _options));
+        auto status = (new (&_bit_shuffle) BitShufflePageDecoder<OLAP_FIELD_TYPE_INT>(_data, _options))->init();
+        if (!status.ok()) return status;
```

Comment:
why not use `RETURN_IF_ERROR()`

Reply:
because too long if write RETURN_IF_ERROR((new (&_bit_shuffle) BitShufflePageDecoder<OLAP_FIELD_TYPE_INT>(_data, _options))->init())